src/btree/bt_sync.c

```diff
@@ -248,6 +266,44 @@ __wt_sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
                     btree->rec_max_txn = mod->rec_max_txn;
                 if (mod != NULL && btree->rec_max_timestamp < mod->rec_max_timestamp)
                     btree->rec_max_timestamp = mod->rec_max_timestamp;
+
+                /*
+                 * Check whether the tree is lookaside btree and verify whether the page contents
+                 * are obsolete or not by checking the visibility of the page stop time pair.
+                 */
+                if (is_las) {
+                    __wt_ref_is_obsolete(session, walk, &obsolete);
+                    if (obsolete) {
+                        /*
+                         * The duplicate tree walk code expects the ref state to be in memory. We
+                         * need to get the duplicate tree walk pointer before marking the current
+                         * tree page as deleted.
+                         */
+                        WT_ERR(__sync_dup_walk(session, walk, flags, &prev));
+
+                        /*
+                         * Mark the page as deleted and also set the parent page as dirty. This is
+                         * to ensure when the parent page is checkpointing, the empty child page
+                         * will be cleaned.
+                         */
+                        previous_state = walk->state;
+                        if (!WT_PAGE_IS_INTERNAL(walk->page)) {
+                            if (!WT_REF_CAS_STATE(session, walk, previous_state, WT_REF_LOCKED))
+                                continue;
+
+                            /* Mark the page as deleted */
+                            WT_REF_SET_STATE(walk, WT_REF_DELETED);
+                            WT_ERR(__wt_page_parent_modify_set(session, walk, true));
+                        }
+
+                        /*
+                         * The Duplicate tree walk pointer is already obtained whenever an obsolete
+                         * page is identified before mark the page as deleted, continue with the
+                         * rest of the tree walk without duplicating walk pointer.
```

Comment:
Thanks! I suggest the following rewording, based on my understanding here:
```
/*
 * The duplicate tree walk pointer, prev, was obtained before we marked the page as deleted.
 * Continue the rest of the tree walk using that pointer.
 */
```

Reply:
Fixed.